cmake/impl_glfw.cmake

```diff
@@ -0,0 +1,16 @@
+if(IMGUI_IMPL_GLFW)
+    hunter_add_package(glfw)
+    find_package(glfw3 REQUIRED)
+
+    add_definitions(-DIMGUI_IMPL_GLFW)
+
+    set(IMGUI_IMPL_DEFINES ${IMGUI_IMPL_DEFINES} -DIMGUI_IMPL_GLFW)
+    set(IMGUI_IMPL_LIBRARIES ${IMGUI_IMPL_LIBRARIES} glfw)
```

Comment:
If you create target `imgui` before including this module, you can do instead:
```cmake
target_compile_definitions(imgui PUBLIC IMGUI_IMPL_GLFW)
target_link_libraries(imgui PUBLIC glfw)
```

Reply:
needed to include headers for the imgui target in the module, is there a way to do both without the lists?

Reply:
Usually I do:
```cmake
option(IMGUI_IMPL_GLFW "..." OFF)

if(IMGUI_IMPL_GLFW)
  list(APPEND IMGUI_HEADERS examples/imgui_impl_glfw.h)
  list(APPEND IMGUI_SOURCE examples/imgui_impl_glfw.cpp)
endif()

add_library(imgui ${IMGUI_HEADERS} ${IMGUI_SOURCE})

if(IMGUI_IMPL_GLFW)
  hunter_add_package(glfw)
  find_package(glfw3 REQUIRED)

  target_compile_definitions(
      imgui
      PUBLIC
      $<BUILD_INTERFACE:IMGUI_IMPL_GLFW>
      IMGUI_IMPL_GLFW
  )

  target_link_libraries(imgui PUBLIC glfw)
endif()
```

You can also try [target_sources](https://cmake.org/cmake/help/latest/command/target_sources.html), but I don't have much experience with it.

Reply:
think I am going to leave as is unless there is a compelling reason the change. Don't much like the idea of splitting it between 2 if's